### ISSUE TITLE
Push-button devices

### ISSUE THREAD

**Issue by justin-vt** [webdeck/homebridge-indigo#18]
Is there any way to do a push-button device (i.e. a device that doesn't have a toggle state?) I'm wondering if there is a HomeKit device-type that might work for this that we could map using homebridge-indigo.

My use case is I want to trigger something in Indigo, but I want it to be stateless so I can trigger it from multiple devices without having to deal with an on/off state.

**Comment by webdeck** [comment]
Indigo action groups map to switches that automatically turn themselves back off, so they essentially act like push buttons.  Is that what you're looking for?

**Comment by jpditto** [comment]
Sorry for dropping this, I ran into another unrelated roadblock with my implementation.

I ended up using two different switches for this - let's call them the "on switch", and the "off switch". If I want to turn the switch on, I turn on the "on switch". If I want to turn the switch off, I turn on the "off switch". I use an action group to turn back off whichever switch has just been turned on, and an internal variable to store the "on/off" status.

**Comment by justin-vt** [comment]
Sorry for dropping this, I ran into another unrelated roadblock with my implementation.

I ended up using two different switches for this - let's call them the "on switch", and the "off switch". If I want to turn the switch on, I turn on the "on switch". If I want to turn the switch off, I turn on the "off switch". I use an action group to turn back off whichever switch has just been turned on, and an internal variable to store the "on/off" status.
